parquet/src/arrow/array_reader/byte_view_array.rs

```diff
@@ -315,26 +315,30 @@ impl ByteViewArrayDecoderPlain {
         let mut read = 0;
         output.views.reserve(to_read);
         while self.offset < self.buf.len() && read != to_read {
-            if self.offset + 4 > self.buf.len() {
-                return Err(ParquetError::EOF("eof decoding byte array".into()));
-            }
-            let len_bytes: [u8; 4] = unsafe {
-                buf.get_unchecked(self.offset..self.offset + 4)
-                    .try_into()
-                    .unwrap()
+            let len_bytes = buf.get(self.offset..self.offset + 4);
+            let len = match len_bytes {
+                None => {
+                    return Err(ParquetError::EOF("eof decoding byte array".into()));
+                }
+                Some(l) => u32::from_le_bytes(l.try_into().unwrap()),
```

Comment:
Is this any more sound? it seems to me this is simply does the same check via using slice 

BTW I think you can write this kind of code more concisely like

```suggestion
            let Some(len_bytes) = buf.get(self.offset..self.offset + 4) else {
                    return Err(ParquetError::EOF("eof decoding byte array".into()));
                };
             let len = u32::from_le_bytes(len_bytes.try_into().unwrap()),
```

Reply:
Yes, I was trying to avoid the version with `let .. else` because I was not sure on the MSRV policies of arrow-rs :-) 

As for being more sound: the original version is unsound if `self.offset + 4` overflows (because it ends up calling .get_unchecked with an empty range, which is UB). In general, the new version achieves the same result without unsafe, which is IMO always a good idea (unless there are big performance issues).

Reply:
This code is also fine IMO - I just wanted to remove the `unsafe`, I attempted to do so without introducing new potential branches but if we believe the compiler to remove them (or they are not perf-critical) then this is indeed the clearest option.